### PR TITLE
patch to remove hack from stable_diffusion.py

### PR DESCRIPTION
**Pull request by segf00lt**
#1805

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/tensor.py                         1               1
------------------------------------------------------------
lines added in the tinygrad folder: 0
```

File: test/test_assign.py [hunk 1]
```diff
@@ -63,5 +64,15 @@ def test_post_permuted_assignment(self):
 
   # TODO: is there a way to sneak in a permute such that it returns the wrong answer?
 
+  def test_cast_assignment(self):
+    a = Tensor(np.arange(N*N, dtype=np.float32)).reshape(N,N)
+    a.realize()
+    oba1 = a.lazydata.output_buffer
+    a.assign(a.cast(dtypes.float64).realize())
```
Comment (wozeparrot) [inline test/test_assign.py:71]: float64 doesn't have full platform support, maybe use one of the int dtypes instead?

Comment (segf00lt) [reply test/test_assign.py:71]: I'll change to int32.

**Comment by geohot** [comment]
Nice find!

**Pull request merged**